xnmt/decoder.py

```diff
@@ -116,7 +120,9 @@ def calc_loss(self, context, ref_action):
       return dy.pickneglogsoftmax(scores, ref_action)
     # minibatch mode
     else:
-      return dy.pickneglogsoftmax_batch(scores, ref_action)
+      ce_loss_term = (1 - self.eps) * (dy.exp(-dy.pickneglogsoftmax_batch(scores, ref_action)))
```

Comment:
This will be much slower than pickneglogsoftmax, so when label smoothing is zero, we should use the normal pickneglogsoftmax.

Reply:
done via the latest commit !